homeassistant/components/sensor/command_line.py

```diff
@@ -81,11 +90,34 @@ def state(self):
         """Return the state of the device."""
         return self._state
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
     def update(self):
         """Get the latest data and updates the state."""
         self.data.update()
         value = self.data.value
 
+        if self._json_attributes:
+            self._attributes = {}
+            if value:
+                try:
+                    json_dict = json.loads(value)
+                    if isinstance(json_dict, collections.Mapping):
+                        self._attributes = {k: json_dict[k] for k in
+                                            self._json_attributes
+                                            if k in json_dict}
+                    else:
+                        _LOGGER.warning("JSON result was not a dictionary")
+                except ValueError:
+                    _LOGGER.warning("Command result could not be parsed as"
+                                    "JSON")
+                    _LOGGER.debug("Erroneous JSON: %s", value)
```

Comment:
Don't double log. Move the value to the warning.

Reply:
Right-o, done, thanks. :)